tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -798,3 +798,62 @@ def test_validation_error_callback_with_malformed_with_invalid_types(workflow_ap
     assert expected_error_code == data['error_code']
     assert expected_message == data['message']
     assert 'errors' in data
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.actions.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.is_pdf_link',
+    return_value=True
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.refextract.extract_references_from_file',
+    return_value=[],
+)
+def test_keep_previously_rejected_now_approved(
```

Comment:
* `test_keep_previously_rejected_from_fully_harvested_category_is_auto_approved`
* what about the opposite case, where it's not from a category that is harvested completely? would it be hard to add such a test asserting it's still rejected?

Reply:
This record that is not fully harvested would be rejected either way since all these kind of records are auto-approved. No?

Reply:
Not everything that is not auto-approved get auto-rejected, some get to the halted (waiting for the match, or for curator approval).